### PR TITLE
Implement `--no-plugins` command arg and config setting

### PR DESCRIPTION
**Pull request by ivan-mogilko**
Add `--no-plugins` command line argument, and related config setting, which tells to **not** load plugins from dynamic libraries.
Built-in plugins and stubs are still going to be used in this case (if they are available).

This may be used to either force engine to use built-in plugins, or as a safety measure to run engine without loading any extra binary.

### CODE REVIEW

File: Engine/plugin/plugin_stubs.cpp [hunk 1]
```diff
@@ -62,11 +62,12 @@ bool RegisterPluginStubs(const char* name)
     ccAddExternalStaticFunction("ShellExecute",                 Sc_PluginStub_Void);
     return true;
   }
-  else if (ags_stricmp(name, "agsappopenurl") == 0)
+  else if (ags_stricmp(name, "agsappopenurl") == 0 ||
+           ags_stricmp(name, "ags_SnowRain20") == 0)
```
Comment (ericoporto) [inline Engine/plugin/plugin_stubs.cpp:66]: I think this is a typo

Comment (ivan-mogilko) [reply Engine/plugin/plugin_stubs.cpp:66]: I put this new line into the wrong condition.

File: Engine/plugin/agsplugin.cpp [hunk 2]
```diff
@@ -1061,47 +1082,31 @@ Engine::GameInitError pl_register_plugins(const std::vector<PluginInfo> &infos)
...
-            AGS::Common::Debug::Printf(kDbgMsg_Info, "Build-in plugin '%s' found and being used.", apl->filename.GetCStr());
-          }
-          else
-          {
-            // Plugin loading has failed at this point, try using built-in plugin function stubs
-            if (RegisterPluginStubs(apl->filename.GetCStr()))
-              AGS::Common::Debug::Printf(kDbgMsg_Info, "Placeholder functions for the plugin '%s' found.", apl->filename.GetCStr());
+            String expect_filename = apl->library.GetFilenameForLib(apl->filename);
+            Debug::Printf(kDbgMsg_Error, "Plugin '%s' could not be loaded (expected '%s'):\n\t%s",
+                apl->filename.GetCStr(), expect_filename.GetCStr());
```
Comment (ericoporto) [inline Engine/plugin/agsplugin.cpp]: Hey, this debug will crash things instantly because it's missing the last parameter.

**Comment by ericoporto** [comment]
I fixed locally the `Debug::Printf` issue and then run and apparently it works, using `--no-plugins` correctly disables the plugins and removing loads them without a problem.

**Pull request merged**